build-scripts/images/parse.py

```diff
@@ -0,0 +1,20 @@
+import yaml
+import sys
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        sys.exit(1)
+    with open(sys.argv[1], "r") as stream:
+        try:
+            manifest = yaml.safe_load(stream)
+            with open(f"{sys.argv[2]}/images-list", "w+") as list:
+                for component in manifest["status"]["components"]:
+                    for asset in component["assets"]:
+                        if asset["name"] == "pause-image":
+                            list.write("PAUSE_IMAGE={}\n".format(asset["image"]["uri"]))
+                        elif asset["name"] == "coredns-image":
+                            list.write("COREDNS_IMAGE={}\n".format(asset["image"]["uri"]))
+                        elif asset["name"] == "metrics-server-image":
+                            list.write("METRICS_SERVER_IMAGE={}\n".format(asset["image"]["uri"]))
+        except yaml.YAMLError as exc:
+            print(exc)
```

Comment:
Adjust the code as you see fit. Reasons for the suggestion:

- Cleanup nested indentation. No need to have nested `with open` blocks, they simply make the code harder to reason about
- sys.argv[2] is already an argument, it should simply the output file.
- load everything in a dict, and simply render the config file in the end.

```suggestion
    eks_manifest_file = sys.argv[1]
    configuration_file = sys.argv[2]

    with open(eks_manifest_file) as fin:
        manifest = yaml.safe_load(fin)

    kvs = {}

    for component in manifest["status"]["components"]:
        for asset in component["assets"]:
            if asset["name"] == "pause-image":
                kvs["PAUSE_IMAGE"] = asset["image"]["uri"]
            elif asset["name"] == "coredns-image":
                kvs["COREDNS_IMAGE"] = asset["image"]["uri"]
            elif asset["name"] == "metrics-server-image":
                kvs["METRICS_SERVER_IMAGE"] = asset["image"]["uri"]

    with open(configuration, "w+") as fout:
        print("\n".join("{}={}".format(k, v) for k, v in kvs.items()))
```